File: client-domain/src/main/java/com/jraska/github/client/users/RepoDetail.java
package com.jraska.github.client.users;

import org.threeten.bp.LocalDateTime;

public class RepoDetail {
  public final RepoHeader header;
  public final Data data;

  public RepoDetail(RepoHeader header, Data data) {
    this.header = header;
    this.data = data;
  }

  public static class Data {
    public final LocalDateTime created;
    public final int issuesCount;
    public final String language;
    public final int subscribersCount;
    public final LocalDateTime pushedAt;

    public Data(LocalDateTime created, int issuesCount, String language, int subscribersCount, LocalDateTime pushedAt) {
      this.created = created;
      this.issuesCount = issuesCount;
      this.language = language;
      this.subscribersCount = subscribersCount;
      this.pushedAt = pushedAt;
    }
  }
}

File: feature/repo/src/main/java/com/jraska/github/client/repo/model/RepoDetail.kt
package com.jraska.github.client.repo.model

import org.threeten.bp.Instant

internal data class RepoDetail(
  val header: RepoHeader,
  val data: Data,
  val pullRequests: PullRequestsState
) {
  class Data(val created: Instant, val issuesCount: Int, val language: String, val subscribersCount: Int)

  sealed class PullRequestsState {
    object Loading : PullRequestsState()
    class PullRequests(val pulls: List<PullRequest>) : PullRequestsState()
    class Error(error: Throwable) : PullRequestsState()
  }

  class PullRequest(val title: String)
}
